filters/map_composition.py

```diff
@@ -79,82 +86,139 @@ def responseComplete(self):
             else:
                 remove_qml = False
 
-            if exists(project_path):
-                if not overwrite:
-                    msg = 'PROJECT is already existing : %s \n' % project_path
-                    request.appendBody(msg)
-                    return
-                else:
-                    remove(project_path)
+            if exists(project_path) and overwrite:
+                # Overwrite means create from scratch again
+                remove(project_path)
+
+            sources_parameters = params.get('SOURCES')
+            # support legacy params: FILES
+            if not sources_parameters:
+                sources_parameters = params.get('FILES')
 
-            files_parameters = params.get('FILES')
-            if not files_parameters:
-                request.appendBody('FILES parameter is missing.\n')
+            if not sources_parameters:
```

Comment:
this is perhaps a matter of coding style.
but sometimes it leads a new developer to believe that there's a repeated *if* condition in the second checks.
perhaps we need to add a comment here describing the business logic. 
something like: 
`# if SOURCES and FILES don't exist, ask for providing SOURCES`

Reply:
Sure, I will add some comment.